jib-cli: init at 0.13.0

sxmair opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->


### Motivation for this addition

Adds jib-cli, a container image builder for Java at version `0.13.0`, to Nixpkgs.

jib-cli enables building optimized OCI and Docker images for Java applications without a Docker daemon, useful for reproducible Java container builds in CI/CD pipelines.

We are using the official prebuilt binary because building from source would require dealing with a complex Gradle setup and many dependencies, making it difficult to package and maintain in Nixpkgs.


###### Notes

Upstream: https://github.com/GoogleContainerTools/jib


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other contributing documentation in corresponding paths.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->
### Tests performed (with reference to "Tested basic functionality" checkbox above):
- [x] Verified result/bin/jib symlink is correct and executable.
- [x]  ./result/bin/jib --version and ./result/bin/jib --help commands running successfully
- [x] Confirmed end-to-end functionality by building a test container image with ./result/bin/jib build --target=docker://jib-cli-quickstart . ([source for testing jib-cli](https://github.com/GoogleContainerTools/jib/blob/master/jib-cli/README.md#quickstart))


---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

sxmair committed:
jib-cli: init at 0.13.0

Signed-off-by: Syed Sumairul Hasan <sumairhasan99@gmail.com>

acid-bong commented on pkgs/build-support/jib-cli/default.nix:
```diff
@@ -0,0 +1,36 @@
+        mv jib-0.13.0/bin/jib $out/bin/
+  '';
+
+  meta = with lib; {
+    description = "Container image builder for Java using Jib CLI";
+    homepage = "https://github.com/GoogleContainerTools/jib";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ ];
+    platforms = platforms.linux;
+  };
```
```suggestion
  meta = with lib; {
    description = "Container image builder for Java using Jib CLI";
    homepage = "https://github.com/GoogleContainerTools/jib";
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [
      # DON'T leave it empty, submit yourself or someone who consented
    ];
    platforms = lib.platforms.linux;
  };
```

acid-bong commented on pkgs/build-support/jib-cli/default.nix:
```diff
@@ -0,0 +1,36 @@
+  pname = "jib-cli";
+  version = "0.13.0";
+
+  src = fetchurl {
+    url = "https://github.com/GoogleContainerTools/jib/releases/download/v${version}-cli/jib-jre-${version}.zip";
+    sha256 = "0q843vrbxlzhb22s8yayc3jm6i9cg4afxf16fa0x7ri1c9va9xp8";
+  };
+
+  nativeBuildInputs = [ unzip ];
+
+  dontBuild = true;
+
+  installPhase = ''
+    
+        mkdir -p $out/bin
+        unzip $src
+        chmod +x jib-0.13.0/bin/jib
+        mv jib-0.13.0/bin/jib $out/bin/
+  '';
```
Why not build it from source? The source is under a free license

acid-bong reviewed (changes requested):
Additionally: 
- read the contributing docs (last checklist item) 
- rename the PR according to the conventions
- and format with Nixfmt only (at least files that you submit, the repo still has unformatted files elsewhere)

ryand56 commented on pkgs/build-support/jib-cli/default.nix:
```diff
@@ -0,0 +1,36 @@
+
+  src = fetchurl {
+    url = "https://github.com/GoogleContainerTools/jib/releases/download/v${version}-cli/jib-jre-${version}.zip";
+    sha256 = "0q843vrbxlzhb22s8yayc3jm6i9cg4afxf16fa0x7ri1c9va9xp8";
```
```suggestion
    hash = "sha256-6PakdmIh5tOBcia47hR5LEVT5WBeeaSFWPDTvvIeBGE=";
```

sxmair committed:
maintainers: add sxmair

Signed-off-by: Syed Sumairul Hasan <sumairhasan99@gmail.com>

ryand56 reviewed (approved):
Can you squash your commits so that the maintainer commit comes first, then the added package?

ryand56 reviewed (commented):
Couple nits, otherwise LGTM

sxmair commented:
@NixOS/nixpkgs-merge-bot merge

acid-bong commented:
OP, merge bot can only operate on autoupdates by r-ryantm, the update bot, and only if called by the package maintainer

Prince213 commented on pkgs/by-name/ji/jib-cli/package.nix:
```diff
@@ -0,0 +1,36 @@
+  dontBuild = true;
+
+  installPhase = ''
+
```
```suggestion
```

sxmair replied:
Hi @Prince213 , 

Building jib-cli from source in nixpkgs would require vendoring a complex gradle build with dozens of dependencies and maintaining it. Given the upstream stability of the official binary and the high effort required, using the official prebuilt binary is more practical in this case. 

Please let me know if adding sourceProvenance will make this acceptable or if a source build is strictly required despite the complexity.

Prince213 reviewed (changes requested):
Can we build this from source?  Otherwise you need to add sourceProvenance.

sxmair commented:
@Prince213 Thanks for the prompt feedback, I've made the necessary adjustments.

sxmair commented:
> > I’ve added it
> 
> Did you? Looks like you've just pushed the same set of changes.

Kindly check again, the changes should be reflected now.

Prince213 commented:
How should be program be run?  It seems stuck on my x86_64-linux machine when running `jib --version`.

sxmair commented:
Hi @Prince213 could you clarify further on this? 

> How should be program be run? It seems stuck on my x86_64-linux machine when running `jib --version`.
> 
> It's recommended that you add some tests, at least a `versionCheckHook`.

If `jib --version` hangs, it may not be suitable for `versionCheckHook`.

Could you help check if `jib help` or `jib --help` exits cleanly on your setup? If so, we can use that for `versionCheckHook`. If those hang too, would it be acceptable to skip `versionCheckHook` in this case, or is there an alternative you recommend? I am unable to run the builds locally on my end and thus, would sincerely appreciate it if you're able to assist with this. 

Thank you!

Prince213 commented:
> Could you help check if `jib help` or `jib --help` exits cleanly on your setup?

No, this also stucks.

> I am unable to run the builds locally on my end

Then this package is not acceptable for Nixpkgs as no one can maintain it.

sxmair commented:
Update: Able to build and test locally and also resolved issue with package.nix for jib-cli.

What was the issue?

- The jib-cli package was building fine but was not runnable because:
  - The binary wrapper was missing, so jib could not execute in the Nix sandbox.
  - jib-cli upstream releases only prebuilt JARs and scripts, requiring explicit handling to invoke the correct Java runtime inside Nix builds.
  - By default, the unpacked binary relies on environment Java, which is not automatically available inside the Nix store runtime environment.

What I changed to make jib-cli runnable
- Added a custom wrapper script (`$out/bin/jib`) that:
  - Uses `${jre}/bin/java` explicitly to run the CLI JAR.
  - Sets the classpath to the unpacked lib/* jars correctly.
  - Passes through all CLI arguments seamlessly.

- Ensured the wrapper is marked executable with the correct shebang.

- Verified that after this change:

  - `./result/bin/jib --version` reports `0.13.0`.
  - `./result/bin/jib --help` and `build` commands work as expected.

Why this is necessary
Without explicitly specifying the Java runtime in the wrapper:
- The binary cannot find Java in the sandbox.
- This leads to “file not found” or “cannot execute” errors.
- This occurs even though the JARs were present and correctly unpacked.

This approach aligns with best practices in Nixpkgs for packaging Java CLI tools, ensuring self-contained, reliable runtime execution.

sxmair commented:
#### Motivation for this addition

Add jib-cli, a container image builder for Java, to Nixpkgs.

jib-cli enables building optimized OCI and Docker images for Java applications without a Docker daemon, useful for reproducible Java container builds in CI/CD pipelines.

##### Things done

- [x] Added jib-cli package at version 0.13.0.
- [x] Binary unpacked using unzip, with correct permissions and phase hooks.
- [x] Defined metadata: description, homepage, license, and maintainer.
- [x] CI confirms successful builds across x86_64-linux, aarch64-linux, x86_64-darwin, and aarch64-darwin.
- [x] Evaluates cleanly.
- [x] Local manual testing performed to confirm build and runtime functionality.

We are using the official prebuilt binary because building from source would require dealing with a complex Gradle setup and many dependencies, making it difficult to package and maintain in Nixpkgs.


##### Tests performed to confirm build and functionality 

- [x] Ran nix-build -A jib-cli, which completed successfully
- [x] Verified result/bin/jib symlink is correct and executable.
- [ ] Confirmed the binary reports the correct version via ./result/bin/jib --version
- [ ] Confirmed CLI help and commands display correctly via ./result/bin/jib --help
- [ ] Confirmed end-to-end functionality by building a test container image with ./result/bin/jib build --target=docker://jib-cli-quickstart . ([source for testing jib-cli](https://github.com/GoogleContainerTools/jib/blob/master/jib-cli/README.md#quickstart))
- [ ] Rebuild stability: Repeated nix-build to confirm deterministic and stable builds.


###### Notes

Upstream: https://github.com/GoogleContainerTools/jib

---

cc reviewers for feedback.
@Prince213

Prince213 commented:
Also you should update your progress in PR description.

Prince213 reviewed (changes requested):
Tested on aarch64-darwin.  Please add versionCheckHook.

sxmair commented:
@Prince213 Done!

Prince213 commented on pkgs/by-name/ji/jib-cli/package.nix:
```diff
@@ -0,0 +1,56 @@
+    runHook postInstall
+  '';
+
+  installCheckPhase = ''
+    echo "Checking installed jib version..."
+    $out/bin/jib --version
+    echo "No additional install checks."
+  '';
```
Remove these, versionCheckHook already does this for you.

sxmair replied:
Hi @Prince213 
for some reason, without these lines stated above, its causing an error during build, which basically expects the path to be "jib-cli" based on `pname`. But if i were to adhere to that, I would have to change the path in line 35 from `$out/bin/jib` to `$out/bin/jib-cli` and i would also have to change the meta.mainProgram from `jib` to `jib-cli` right? And for this maybe you can advice if that is okay (changing jib to jib-cli for path and meta.mainProgram).

error log when i run `nix-build -A jib-cli` when i remove lines 40-44.

```
➜  new-nixpkgs git:(add-jib-cli) ✗ nix-build -A jib-cli                                              
this derivation will be built:
  /nix/store/9nqqbbw31rxg8zih1vislhiihvh9acpj-jib-cli-0.13.0.drv
building '/nix/store/9nqqbbw31rxg8zih1vislhiihvh9acpj-jib-cli-0.13.0.drv'...
Using versionCheckHook
Running phase: unpackPhase
unpacking source archive /nix/store/jhv50flcblsjjk1yd2jmccfpcwkl5m84-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: installPhase
Running phase: fixupPhase
stripping java archives in /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/failureaccess-1.0.1.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/failureaccess-1.0.1.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jib-plugins-extension-common-0.2.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jib-plugins-extension-common-0.2.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jackson-annotations-2.15.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jackson-annotations-2.15.2.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-lang3-3.14.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-lang3-3.14.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jib-cli-0.13.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jib-cli-0.13.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/checker-qual-3.33.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/checker-qual-3.33.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-text-1.10.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-text-1.10.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/snakeyaml-2.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/snakeyaml-2.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/error_prone_annotations-2.18.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/error_prone_annotations-2.18.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jackson-dataformat-yaml-2.15.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jackson-dataformat-yaml-2.15.2.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/grpc-context-1.27.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/grpc-context-1.27.2.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/google-auth-library-credentials-1.10.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/google-auth-library-credentials-1.10.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/google-auth-library-oauth2-http-1.10.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/google-auth-library-oauth2-http-1.10.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/guava-32.1.2-jre.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/guava-32.1.2-jre.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/google-http-client-gson-1.42.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/google-http-client-gson-1.42.2.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/google-http-client-apache-v2-1.42.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/google-http-client-apache-v2-1.42.2.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/httpclient-4.5.13.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/httpclient-4.5.13.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jackson-databind-2.15.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jackson-databind-2.15.2.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-compress-1.26.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-compress-1.26.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/google-http-client-1.42.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/google-http-client-1.42.2.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/j2objc-annotations-2.8.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/j2objc-annotations-2.8.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/auto-value-annotations-1.9.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/auto-value-annotations-1.9.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/opencensus-api-0.31.1.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/opencensus-api-0.31.1.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/opencensus-contrib-http-util-0.31.1.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/opencensus-contrib-http-util-0.31.1.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/gson-2.9.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/gson-2.9.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jsr305-3.0.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jsr305-3.0.2.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/picocli-4.7.4.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/picocli-4.7.4.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-logging-1.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-logging-1.2.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jib-plugins-common-0.1.0-SNAPSHOT.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jib-plugins-common-0.1.0-SNAPSHOT.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-codec-1.11.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-codec-1.11.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/httpcore-4.4.15.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/httpcore-4.4.15.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-io-2.15.1.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/commons-io-2.15.1.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jib-build-plan-0.4.0.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jib-build-plan-0.4.0.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jackson-core-2.15.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jackson-core-2.15.2.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/asm-9.5.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/asm-9.5.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jib-core-0.27.1-SNAPSHOT.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jib-core-0.27.1-SNAPSHOT.jar
stripping java archive /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jackson-datatype-jsr310-2.15.2.jar
Normalized /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib/jackson-datatype-jsr310-2.15.2.jar
shrinking RPATHs of ELF executables and libraries in /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0
checking for references to /build/ in /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0...
patching script interpreter paths in /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0
stripping (with command strip and flags -S -p) in  /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/lib /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/bin
Running phase: installCheckPhase
Executing versionCheckPhase
versionCheckHook: /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0/bin/jib-cli was not found, or is not an executable
error: Cannot build '/nix/store/9nqqbbw31rxg8zih1vislhiihvh9acpj-jib-cli-0.13.0.drv'.
       Reason: builder failed with exit code 2.
       Output paths:
         /nix/store/m5gkl5ysvr7j20hsqnnacf0mw0fpcpl5-jib-cli-0.13.0

```

sxmair replied:
Another alternative i found if i were to not specify the installCheckPhase, would be to create another symlink to comply with the jib-cli path.

by adding `ln -s $out/bin/jib $out/bin/jib-cli` right after line 36.  this way i dont have to explicity specify the `installCheckPhase` as well.

Do let me know what works best. thank you!

Prince213 replied:
RTFM https://nixos.org/manual/nixpkgs/unstable/#versioncheckhook

Prince213 commented on pkgs/by-name/ji/jib-cli/package.nix:
```diff
@@ -0,0 +1,52 @@
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out
+    cp -r bin lib $out/
```
Do we need this?  It will copy jib and jib.bat, but no one will be used.

sxmair replied:
@Prince213 If we exclude copying /bin over, only the `jib` will be generated. Is it safe to assume the `jib.bat` isnt required in nix?

Prince213 replied:
That's for Windows so I don't think it's necessary.

sxmair replied:
Noted, will disable the copying over of `/bin` folder then.

sxmair commented:
@Prince213 thanks for precaution. I am confident that the addition of this package satisfies all the criterias listed in the abovementioned link.

- **General usability** : `jib-cli` is a mature and widely-used CLI tool for building container images, maintained by Google under the [Jib project](https://github.com/GoogleContainerTools/jib).
- **License**: Licensed under Apache 2.0 as [following](https://github.com/GoogleContainerTools/jib/blob/master/LICENSE).
- **Usage and ecosystem**: `jib` is broadly used in many CI/CD workflows and Java-based container build pipelines (it also has 14k stars on GitHub).
- **Upstream maintenance**: Upstream is actively maintained with recent commits and releases, with the GitHub issues and PRs being handled regularly.
- **Willing to maintain**: Yes, I'm listed as a maintainer and am committed to supporting this package for at least one complete Nixpkgs release life-cycle. If I become unavailable, I will ensure a suitable replacement is found.

SuperSandro2000 commented on pkgs/by-name/ji/jib-cli/package.nix:
```diff
@@ -0,0 +1,52 @@
+    mkdir -p $out
+    cp -r lib $out/
+    makeWrapper ${jre}/bin/java $out/bin/jib \
+      --add-flags "-cp $(find $out/lib -name '*.jar' | tr '\n' ':') com.google.cloud.tools.jib.cli.JibCli"
```
Can't we hardcode the binaries here instead of finding them on every startup?

sxmair commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 421200`
Commit: `bc2146534ec5d5deebee291e3f429418df612087`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>jib-cli</li>
  </ul>
</details>

sxmair commented:
Thanks alot @SuperSandro2000 and everyone else for the immense help on this PR! 😄
